File: frontend/src/utils/csvExport.js
export const BRAND_NAME = 'PathyCode';
export const COMPANY_TAGLINE = 'Business & financial operations';

const escapeCell = (value) => `"${String(value ?? '').replace(/"/g, '""')}"`;

/**
 * Build CSV text with branded header rows (company name, tagline, title, description).
 */
export function buildBrandedCsvContent(exportTitle, description, headers, rows) {
  const generated = new Date().toLocaleString();
  const brandLines = [
    ['Company', BRAND_NAME],
    ['Tagline', COMPANY_TAGLINE],
    ['Export', exportTitle],
    description ? ['Description', description] : null,
    ['Generated', generated],
    [],
  ]
    .filter((line) => line !== null)
    .map((line) => (line.length === 0 ? '' : line.map(escapeCell).join(',')));

  const dataLines = [
    headers.map(escapeCell).join(','),
    ...rows.map((row) => row.map(escapeCell).join(',')),
  ];

  return [...brandLines, ...dataLines].join('\r\n');
}

/**
 * Trigger a branded CSV file download in the browser.
 */
export function downloadBrandedCsv(filename, exportTitle, description, headers, rows) {
  const BOM = '\uFEFF';
  const csvContent = buildBrandedCsvContent(exportTitle, description, headers, rows);
  const blob = new Blob([BOM + csvContent], { type: 'text/csv;charset=utf-8;' });
  const url = URL.createObjectURL(blob);
  const link = document.createElement('a');
  link.href = url;
  link.download = filename;
  link.style.display = 'none';
  document.body.appendChild(link);
  link.click();
  document.body.removeChild(link);
  setTimeout(() => URL.revokeObjectURL(url), 200);
}

File: PathyCodeback/csv_export.py
"""
Branded CSV export helpers — company name, tagline, and document context.
"""
import csv

from django.conf import settings
from django.http import HttpResponse
from django.utils import timezone


def get_brand_meta():
    """
    Read company branding strings from Django settings.

    Returns:
        dict: ``company_name`` and ``tagline`` keys populated from
        ``BRAND_NAME`` and ``COMPANY_TAGLINE`` (with sensible defaults).
    """
    return {
        'company_name': getattr(settings, 'BRAND_NAME', 'PathyCode'),
        'tagline': getattr(settings, 'COMPANY_TAGLINE', 'Business & financial operations'),
    }


def write_branded_csv_header(writer, export_title, description=None):
    """
    Write company branding and export metadata rows before column headers.

    Args:
        writer: ``csv.writer`` instance bound to the HTTP response or buffer.
        export_title (str): Human-readable export name (e.g. ``Clients Export``).
        description (str, optional): Short summary shown on the second metadata row.
    """
    brand = get_brand_meta()
    writer.writerow(['Company', brand['company_name']])
    if brand.get('tagline'):
        writer.writerow(['Tagline', brand['tagline']])
    writer.writerow(['Export', export_title])
    if description:
        writer.writerow(['Description', description])
    writer.writerow(['Generated', timezone.now().strftime('%Y-%m-%d %H:%M')])
    writer.writerow([])


def branded_csv_response(filename, export_title, description=None):
    """
    Create an ``HttpResponse`` with a UTF-8 BOM and branded CSV header.

    Args:
        filename (str): Download filename (e.g. ``clients.csv``).
        export_title (str): Title embedded in the CSV header block.
        description (str, optional): Optional description row in the header block.

    Returns:
        tuple: ``(HttpResponse, csv.writer)`` — response is ready for data rows.
    """
    response = HttpResponse(content_type='text/csv; charset=utf-8')
    response['Content-Disposition'] = f'attachment; filename="{filename}"'
    response.write('\ufeff')
    writer = csv.writer(response)
    write_branded_csv_header(writer, export_title, description)
    return response, writer
